### PR TITLE
Renaming transport_reconnect to retryUtils 

### PR DESCRIPTION
**Pull request by sukhmanm**
*Description of changes:* Renaming all `transport_reconnect` utilities to `retryUtils` to avoid strict association to transport reconnection.

**General renaming pattern:**
- `transport_reconnect*` -> `retry_utils*`
- `TransportReconnectParams_t` -> `RetryUtilsParams_t`
- `Transport_Reconnect*` -> `RetryUtils_*`

`transport_reconnect_posix.c` moved from platform/posix/transport/src to platform/posix (as `retry_utils_posix.c`)
`transportFilePaths.cmake` moved from platform/posix/transport to platform/posix (as `posixFilePaths.cmake`)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: platform/include/retry_utils.h [hunk 1]
```diff
@@ -40,49 +40,48 @@
...
+#define MAX_RETRY_ATTEMPTS               4U
 
 /* @brief Initial fixed backoff value in seconds between two successive
- * connects. A random jitter value is added to every backoff value  */
-#define INITIAL_RECONNECT_BACKOFF_SECONDS    1U
+ * retries. A random jitter value is added to every backoff value  */
+#define INITIAL_RETRY_BACKOFF_SECONDS    1U
 /* @brief Max backoff value in seconds */
-#define MAX_RECONNECT_BACKOFF_SECONDS        128U
+#define MAX_RETRY_BACKOFF_SECONDS        128U
```
Comment (yourslab) [inline platform/include/retry_utils.h:51]: Thanks for renaming the macros as well

File: platform/include/retry_utils.h [hunk 2]
```diff
@@ -20,13 +20,13 @@
...
- * @brief Declaration of the exponential backoff reconnect logic utility functions
+ * @file retry_utils.h
+ * @brief Declaration of the exponential backoff retry logic utility functions
  * and constants.
  */
 
-#ifndef TRANSPORT_RECONNECT_H_
-#define TRANSPORT_RECONNECT_H_
+#ifndef RETRY_UTILS__H_
+#define RETRY_UTILS__H_
```
Comment (yourslab) [inline platform/include/retry_utils.h:29]: Double underscore in the header include guard

File: platform/posix/posixFilePaths.cmake [hunk 3]
```diff
@@ -7,21 +7,24 @@
...
 set( COMMON_TRANSPORT_INCLUDE_PUBLIC_DIRS
-     ${CMAKE_CURRENT_LIST_DIR}/include
+     ${CMAKE_CURRENT_LIST_DIR}/transport/include
+     ${PLATFORM_DIR}/include )
+
+# Retry logic source files.
+set( RETRY_SOURCES
+     ${CMAKE_CURRENT_LIST_DIR}/retry_utils_posix.c )
+# Retry Public Include directories.
+set( COMMON_RETRY_INCLUDE_PUBLIC_DIRS
```
Comment (yourslab) [inline platform/posix/posixFilePaths.cmake:29]: I feel like this shouldn't be prefixed with `COMMON_` anymore.

Comment (sukhmanm) [reply platform/posix/posixFilePaths.cmake:29]: Good point. I've updated it, thanks.

**Review by yourslab — DISMISSED** [review]

**Review by yourslab — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**